src/main/java/org/magicdgs/readtools/tools/mapped/ComputeProperStatByWindow.java

```diff
@@ -0,0 +1,209 @@
+            + "\n\n"
+            + "Find more information about this tool in "
+            + RTHelpConstants.DOCUMENTATION_PAGE + "ComputeProperStatByWindow.html";
+
+
+
+    /**
+     * Tab-delimited output file with the statistic over the windows. A header defines the order of
+     * each statistic and the first column the window in the form contig:start-end.
+     */
+    @Argument(fullName = RTStandardArguments.OUTPUT_LONG_NAME, shortName = RTStandardArguments.OUTPUT_SHORT_NAME, doc = "Tab-delimited output file with the statistic over the windows")
+    public String outputArg;
+
+    @Argument(fullName = RTStandardArguments.FORCE_OVERWRITE_NAME, shortName = RTStandardArguments.FORCE_OVERWRITE_NAME, doc = RTStandardArguments.FORCE_OVERWRITE_DOC, optional = true, common = true)
+    public Boolean forceOverwrite = false;
+
+    // name for the contig argument
+    private static final String WINDOW_CONTIG_NAME = "contig";
+    /**
+     * Limit the computation to the provided contig(s). This argument is used instead of
+     * interval arguments and might be removed in the future if intervals are supported.
+     */
+    @Argument(fullName = WINDOW_CONTIG_NAME, doc = "Limit the computation to the provided contig(s)", optional = true)
+    public List<String> contig = new ArrayList<>();
+
+    // TODO: support sliding-window (depends on the engine supporting it)
+    @Argument(fullName = "window-size", doc = "Window size to perform the analysis")
+    public Integer window;
+
+    // TODO: this should be a plugin (https://github.com/magicDGS/ReadTools/issues/448)
+    @ArgumentCollection
+    public ComputeProperStatByWindowArgs engineArgs = new ComputeProperStatByWindowArgs();
+
+
+    @Override
+    public List<ReadFilter> getDefaultReadFilters() {
+        // using the mapped filter speeds-up processing
+        // TODO: should we also add WellFormedReadFilter?
+        return Arrays.asList(ReadFilterLibrary.MAPPED, ReadFilterLibrary.PRIMARY_LINE);
+    }
+
+    // engine for computing the stats
+    private ProperStatWindowEngine engine;
+
+    @Override
+    public void onTraversalStart() {
+        // first check if the path exits or not
+        final Path path = IOUtils.getPath(outputArg);
+        if (!forceOverwrite && Files.exists(path)) {
+            throw new RTUserExceptions.OutputFileExists(outputArg);
+        }
+
+        // TODO: get rid of this limitation (requires engine changes)
+        if (getHeaderForReads().getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
+            throw new UserException(getToolName() + " only supports coordinate-sorted inputs.");
+        }
+
+        // validate that there is a sequence dictionary
+        final SAMSequenceDictionary dictionary = getBestAvailableSequenceDictionary();
+        if (dictionary == null) {
+            throw new UserException(getToolName() + " requires a sequence dictionary.");
```

Comment:
If sort order is "coordinate", there is always a dictionary - thus, this check is unneeded.

Reply:
Removed.